subsys/net/ip/connection.c

```diff
@@ -248,6 +248,12 @@ static struct net_conn *conn_find_handler(uint16_t proto, uint8_t family,
 			continue;
 		}
 
+#if defined(CONFIG_NET_CONTEXT_REUSEADDR)
+		if (conn->context->options.reuseaddr != 0) {
```

Comment:
Is this actually correct? We're checking the option on a socket (net_context) that has already registered a connection, and not the one that that tries to register a new connection (and should have SO_REUSEADDR set)?

And to me, it'd make a bit more sense to let the `conn_find_handler()` function work as it is, and leave the decistion on what to do with an already existing connection to `net_conn_register()` (i. e. check whether the flag is set).

Reply:
You are right. This is fixed (and further improved) in the new commit